proposals/NightlyBuilds.md

```diff
@@ -0,0 +1,49 @@
+# Cordova nightly builds proposal
+
+We'd like to complete the work, started by @stevesgill in [CB-7904](https://issues.apache.org/jira/browse/CB-7904) and get nightly builds of cordova and related packages published on NPM.
+
+## Background
+
+Currently the only way to test the development version of cordova is to clone cordova and all dependent repos, link them to each other and npm install every cordova repository, which is pretty complicated. Using `cordova-coho` tool makes the process of setting up development cordova version a bit easier, but there is still a lot of caveats.
+
+Having development version of cordova published on npm would allow everyone to easily set up development version of cordova to test unreleased features, reproduce the bugs or confirm they get fixed in current version.
+
+Nightly builds in our understanding are an npm packages created from current 'master' branches of respective repos and available on NPM under `nightly` dist tag, so they easily could be installed like `npm i cordova@nightly`. All appropriate dependencies of nightly packages also will be updated to point to their nightly versions, e.g. `cordova-cli@6.2.0-nightly` will depend on `cordova-lib@6.2.0-nightly`, etc.
+
+The nightly version of cordova-lib will also has main platform pinned to their nightly versions, so `cordova install android` will automatically install `cordova-android@nightly` to allow early adopters test new features and bugfixes in platforms.
+
+The full format of nightly version is `<package_id>@<next_version>-nightly.<build_date>+<commit_SHA>`, for example
+
+```
+cordova-lib@6.2.0-nightly.2016.05.12+1649800352c51e832262b8469cb4c7f362de28ec
+```
+
+This format is fully Semver-compatible and allows to us to easily recreate the same the cordova setup locally
+
+The nightly packages, we're planning to release, include cordova-cli, cordova-lib and the major platforms (Android, iOS, Windows)
+
+## How it will work
+
+All the aspects of creating and publishing nightly builds are automated using `cordova coho` command. Running `coho nightly` assumes the following steps:
+
+1. For every package to publish update respective repo to the latest commit
+2. For every package update version to nightly version
+  - Also update dependencies versions to point to nightly versions of other `cordova-*` dependencies
+3. For every package run the tests to ensure that release will not contain any known bugs

```

Comment:
> does it imply the nightly build will not be published

I think this is the right approach. The tests has already discovered some issues, that would prevent nightly builds to work properly (see [CB-11261](https://issues.apache.org/jira/browse/CB-11261) and https://github.com/apache/cordova-lib/pull/440), so it might be a good idea to avoid publishing nightly if the tests are failing.